### PR TITLE
Add https support

### PR DESCRIPTION
**Pull request by ditac**
*Issue #29 

*Description of changes:*
Adding https support.
Pending issues - #30 , #31 , #32 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java [hunk 1]
```diff
@@ -60,44 +77,103 @@ public void run() {
...
+            Security.addProvider(new BouncyCastleProvider());
             HttpServer server = null;
-            if (bindHost != null && !bindHost.trim().isEmpty()) {
-                LOG.info("Binding to Interface: {}", bindHost);
-                server = HttpServer.create(new InetSocketAddress(InetAddress.getByName(bindHost.trim()), readerPort),
-                                           INCOMING_QUEUE_LENGTH);
-            } else {
-                LOG.info("Value Not Configured for: {} Using default value: binding to all interfaces", WEBSERVICE_BIND_HOST_NAME);
-                server = HttpServer.create(new InetSocketAddress(readerPort), INCOMING_QUEUE_LENGTH);
+            if(PluginSettings.instance().getHttpsEnabled()) {
```
Comment (yeajung) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java]: nit: space between "if" and "("

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java [hunk 2]
```diff
@@ -60,44 +77,103 @@ public void run() {
...
             LOG.error("Port  {} is already in use...exiting", readerPort);
             Runtime.getRuntime().halt(1);
-        } catch(Exception ex) {
+        } catch (Exception ex) {
             LOG.error("Exception in starting Reader Process: " + ex.toString());
             Runtime.getRuntime().halt(1);
         }
     }
 
+    private static HttpServer createHttpsServer(int readerPort) throws Exception {
```
Comment (yeajung) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java]: should the return type be HttpsServer?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/CertificateUtils.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        String keyFilePath = PluginSettings.instance().getSettingValue(PRIVATE_KEY_FILE_PATH);
+        PrivateKey pk = getPrivateKey(new FileReader(keyFilePath));
+        KeyStore ks = emptyStore();
+        Certificate certificate = getCertificate(new FileReader(certFilePath));
+        ks.setCertificateEntry(ALIAS_CERT, certificate);
+        ks.setKeyEntry(ALIAS_PRIVATE, pk, IN_MEMORY_PWD.toCharArray(), new Certificate[]{certificate});
+        return ks;
+    }
+
+    public static KeyStore emptyStore() throws Exception {
```
Comment (sendkb) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/CertificateUtils.java]: minor: s/emptyStore/createAndReturnEmptyStore()

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java (hunk 2 above)
Comment (sendkb) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java]: HttpsServer is a subclass of HttpServer.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java [hunk 4]
```diff
@@ -58,6 +60,22 @@ public String getSettingValue(String settingName, String defaultValue) {
         return settings.getProperty(settingName, defaultValue);
     }
 
+    private static void loadHttpsEnabled(PluginSettings instance, Properties settings) throws Exception {
+        String httpsEnabledString = settings.getProperty(HTTPS_ENABLED, "False");
+        if ( httpsEnabledString == null ) {
```
Comment (sendkb) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java]: nit: remove extra space between '(' and httpsEnabledString

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/CertificateUtils.java (hunk 3 above)
Comment (ditac) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/CertificateUtils.java]: Done.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java (hunk 1 above)
Comment (ditac) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java]: Done

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java (hunk 4 above)
Comment (ditac) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java]: Done

**Pull request merged**